test/parallel/test-snapshot-cwd.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+// This tests that user land snapshots works when the instance restored from
+// the snapshot is launched with -p and -e
+
+require('../common');
+const assert = require('assert');
+const { spawnSync } = require('child_process');
```

Comment:
Can you replace it with `const { spawnSyncAndExitWithoutError } = require('../common/child_process');` and use it below? That would log additional information if the child process does not exit in the way expected which is important for debugging the CI. You can check out the `test/common/README.md` to see what it is. For the first test case I think it just needs an empty option as the last argument and the second one it can use `{trim: true, stdout: subdir}`